src/handlers/decision.rs

```diff
@@ -0,0 +1,340 @@
+    interactions::ErrorComment,
+};
+use anyhow::bail;
+use anyhow::Context as Ctx;
+use chrono::{DateTime, Duration, Utc};
+use parser::command::decision::Resolution;
+use parser::command::decision::*;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+pub const DECISION_PROCESS_JOB_NAME: &str = "decision_process_action";
+
+#[derive(Serialize, Deserialize)]
+pub struct DecisionProcessActionMetadata {
+    pub message: String,
+    pub get_issue_url: String,
+    pub status: Resolution,
+}
+
+pub(super) async fn handle_command(
+    ctx: &Context,
+    _config: &DecisionConfig,
+    event: &Event,
+    cmd: DecisionCommand,
+) -> anyhow::Result<()> {
+    let db = ctx.db.get().await;
+
+    let DecisionCommand {
+        resolution,
+        reversibility,
+        team: team_name,
+    } = cmd;
+
+    let issue = event.issue().unwrap();
+    let user = event.user();
+
+    match get_issue_decision_state(&db, &issue.number).await {
+        Ok(_state) => {
+            // TO DO
+            let cmnt = ErrorComment::new(
+                &issue,
+                "We don't support having more than one vote yet. Coming soon :)",
+            );
+            cmnt.post(&ctx.github).await?;
+
+            Ok(())
+        }
+        _ => {
+            let is_team_member = user.is_team_member(&ctx.github).await.unwrap_or(false);
+            if !is_team_member {
+                let cmnt = ErrorComment::new(
+                    &issue,
+                    "Only team members can be part of the decision process.",
+                );
+                cmnt.post(&ctx.github).await?;
+
+                return Ok(());
+            }
+
+            match team_name {
+                None => {
+                    let cmnt = ErrorComment::new(
+                        &issue,
+                        "In the first vote, is necessary to specify the team name that will be involved in the decision process.",
+                    );
+                    cmnt.post(&ctx.github).await?;
+
+                    Ok(())
+                }
+                Some(team_name) => {
+                    match github::get_team(&ctx.github, &team_name).await {
+                        Ok(Some(team)) => {
+                            let start_date: DateTime<Utc> = chrono::Utc::now().into();
+                            let end_date: DateTime<Utc> =
+                                start_date.checked_add_signed(Duration::days(10)).unwrap();
+
+                            let mut current: BTreeMap<String, Option<UserStatus>> = BTreeMap::new();
+                            let mut history: BTreeMap<String, Vec<UserStatus>> = BTreeMap::new();
+
+                            // Add team members to current and history
+                            for member in team.members {
+                                current.insert(member.github.clone(), None);
+                                history.insert(member.github.clone(), Vec::new());
```

Comment:
Maintaining history by hand is going to be very tricky -- can we do this in some more automated way? For example, just before writing new state to the DB query the previous state and write that to history as part of the change?

Reply:
For now, we just implement the support for a first comment, we actually query the db and if there's a previous state we should do what you mention. This is the initialization for a new entry in the `issue_decision_state` table.